Lib/argparse.py

```diff
@@ -1667,22 +1667,23 @@ def _get_optional_kwargs(self, *args, **kwargs):
                 raise ValueError(
                     f'invalid option string {option_string!r}: '
                     f'must start with a character {self.prefix_chars!r}')
-
-            # strings starting with two prefix characters are long options
             option_strings.append(option_string)
-            if len(option_string) > 1 and option_string[1] in self.prefix_chars:
-                long_option_strings.append(option_string)
 
         # infer destination, '--foo-bar' -> 'foo_bar' and '-x' -> 'x'
         dest = kwargs.pop('dest', None)
         if dest is None:
-            if long_option_strings:
-                dest_option_string = long_option_strings[0]
-            else:
-                dest_option_string = option_strings[0]
-            dest = dest_option_string.lstrip(self.prefix_chars)
+            short_option_dest = None
+            for option_string in option_strings:
+                if len(option_string) > 2:
```

Comment:
I don't think we can handle options like this because this will introduce a regression for double-dashed options. For example, `add_argument('-f', '-foo', '--foobar')` should set `dest=foobar` but now this will set `dest` to the first argument that matches (i.e. `foo`). 

I think we want double-dashed to take first priority, then single-dashed long options, then short options. 

That said, we probably also want a test to catch this.